cauldron/es.py

```diff
@@ -0,0 +1,240 @@
+# SUNNY SINGH 13/07/2018
+#this module is written in mind keeping in mind ES version 6.2
+#reason for your aiohttp requests is to  be able to upgrade to higher version of
+#elastic search without having to depend on version of aioes and aiohttp
+import logging, aiohttp, json
+from asyncio import coroutine, async
+logger = logging.getLogger(__name__)
+
+
+def es_old_new_mapper(data):
+    res = {}
+    if not data:
+        return data
+    if isinstance(data, list):
+        res = []
+        res = [ es_old_new_mapper(dt) for dt in data ]
+        return res
+    elif isinstance(data, dict):
+        for key,val in data.items():
+            if key == "type" and isinstance(val, str) and val == "string":
```

Comment:
This will convert all instances of `"type": "string"` and `"index": "not_analyzed"` to `"type": "text"`, when it ideally should be `"type": "keyword"` because the field isn't being analysed.

Reply:
will test and release later.